internal/inc/file_internal.h

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#ifndef FILE_INTERNAL_H
+#define FILE_INTERNAL_H
+
+#include "umock_c/umock_c_prod.h"
+#include "constbuffer.h"
+#include "file.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct FILE_WRITE_DATA_TAG
+{
+    FILE_WRITE_CB user_callback;
+    void* user_context;
+    CONSTBUFFER_HANDLE source;
+}FILE_WRITE_DATA;
+
+typedef struct FILE_READ_DATA_TAG
+{
+    FILE_READ_CB user_callback;
+    void* user_context;
+    uint32_t size;
+    unsigned char* destination; /*where the readfile reads*/
+}FILE_READ_DATA;
+
+typedef union FILE_IO_DATA_TAG
+{
+    FILE_READ_DATA read_data;
+    FILE_WRITE_DATA write_data;
+}FILE_IO_DATA;
+
+#define FILE_ASYNC_OPERATION_VALUES \
+    FILE_ASYNC_WRITE, \
+    FILE_ASYNC_READ
+
+MU_DEFINE_ENUM(FILE_ASYNC_OPERATION, FILE_ASYNC_OPERATION_VALUES)
+MU_DEFINE_ENUM_STRINGS(FILE_ASYNC_OPERATION, FILE_ASYNC_OPERATION_VALUES)

```

Comment:
>MU_DEFINE_ENUM_STRINGS [](start = 0, length = 22)

Belongs in .c file, not header

Reply:
Do I even need to do this given that I am defining my own INVALID_ARGS enum value?

---
In reply to: [448501146](https://github.com/Azure/azure-c-pal/pull/7#discussion_r448501146) [](ancestors = 448501146)